docs/build/html/S3Storage.html

```diff
@@ -0,0 +1,296 @@
+
+<span class="n">s3_client</span> <span class="o">=</span> <span class="n">boto3</span><span class="o">.</span><span class="n">client</span><span class="p">(</span><span class="s1">&#39;s3&#39;</span><span class="p">,</span> <span class="o">**</span><span class="n">credentials</span><span class="p">)</span>
+<span class="n">s3_client</span><span class="o">.</span><span class="n">upload_file</span><span class="p">(</span>
+    <span class="n">Filename</span><span class="o">=</span><span class="s1">&#39;/path/to/local/file,</span>
+    <span class="n">Bucket</span><span class="o">=</span><span class="s1">&#39;my-external-synapse-bucket&#39;</span><span class="p">,</span>
+    <span class="n">Key</span><span class="o">=</span><span class="s1">&#39;path/within/bucket/file&#39;</span><span class="p">,</span>
+<span class="p">)</span>
+</pre></div>
+</div>
+</div></blockquote>
+</div>
+<div class="section" id="automatic-transfers-to-from-sts-storage-locations-using-boto3-with-synapseclient">
+<h3>Automatic transfers to/from STS storage locations using boto3 with synapseclient<a class="headerlink" href="#automatic-transfers-to-from-sts-storage-locations-using-boto3-with-synapseclient" title="Permalink to this headline">¶</a></h3>
+<p>The Python Synapse client can be configured to automatically use STS tokens to perform uploads and downloads to enabled
+storage locations using an installed boto3 library rather than through the traditional Synapse client APIs.
+This can improve performance in certain situations, particularly uploads of large files, as the data transfer itself
+can be conducted purely against the AWS S3 APIs, only invoking the Synapse APIs to retrieve the necessary token and
+to update Synapse meta data in the case of an upload. Once configured to do so, retrieval of STS tokens for supported
+operations occurs automatically without any change in synapseclient usage.</p>
+<p>To enable STS/boto3 transfers, do the following:</p>
+<ol class="arabic simple">
+<li><p>Ensure that boto3 is installed in the same Python installation as synapseclient.</p></li>
+</ol>
+<blockquote>
+<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">boto3</span>
+</pre></div>
+</div>
+</div></blockquote>
+<ol class="arabic simple" start="2">
+<li><p>Update your Synapse client configuration file (typically “.synapseConfig” in your $HOME directory unless otherwise
+configured) with the following (adding the [transfer] section if it is not already present), or configure
+it on a per Synapse client object basis by setting the <strong>use_boto_sts_transfers</strong> property.</p></li>
```

Comment:
```suggestion
<li><p>To enable automatic transfers on all uploads and downloads, update your Synapse client configuration file (typically “.synapseConfig” in your $HOME directory, unless otherwise
configured) with the [transfer] section, if it is not already present. To leverage STS/boto3 transfers on a per Synapse client object basis, set the <strong>use_boto_sts_transfers</strong> property.</p></li>
```